hashintel/hash
# Fix page functionality in sidebar

## 🌟 What is the purpose of this PR?




This PR fixes various functionality in the sidebar of the HASH workspace, including:
- creating pages
- creating nested pages
- rearranging pages
- archiving pages

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1203079430924743/f) _(internal)_

## 🔍 What does this change?




- replaces the existing `setParentPage` and `createPage` mutations with updated implementations
- adds a `updateKnowledgePage` mutation
- replaces the existing `accountPages` with a `knowledgePages` query
- excludes the GraphQL integration tests from lint checks so that these can be addressed in follow-up

## 📜 Does this require a change to the docs?



No.

## 🐾 Next steps



- [Re-implement GraphQL integration tests](https://app.asana.com/0/1202805690238892/1203084714149810/f)

## 🛡 What tests cover this?



None, the GQL integration tests [will need to be rewritten in follow-up](https://app.asana.com/0/1202805690238892/1203084714149810/f). This PR has to be tested manually.

## ❓ How to test this?



1. Checkout this branch
2. Run `yarn external-services up` locally
3. Run `yarn dev`
4. Open `localhost:3000` and login
5. You should be redirected to the HASH workspace, where you can use the sidebar to create/rearrange/archive pages.

## 📹 Demo




https://user-images.githubusercontent.com/42802102/193824940-2f6226b7-5b2e-4c49-9e81-af6a7ab32320.mov
